components/sysutils/p0f/patches/001-p0f-3.09b-solarisfix.patch

```diff
@@ -0,0 +1,859 @@
++	add the privilege set for low-level networking and delegate the shared
++	NIC's device node, for example:
++		# zonecfg -z mailin << EOF
++		set limitpriv=default,priv_net_rawaccess
++		add device
++		  set match=/dev/e1000g0
++		  end
++		verify
++		commit
++		exit
++		EOF
++
++	No reconfiguration is needed for an exclusive-IP zone nor the global
++	zone, this service manifest includes all the needed privileges for
++	the service (statement valid as of Solaris 10u10 release).
++-->
++<service_bundle type='manifest' name='export'>
++  <service name='network/p0f-daemon' type='service' version='0'>
++    <create_default_instance enabled='false'/>
++    <single_instance/>
++    <dependency name='fs-local' grouping='require_all' restart_on='none' type='service'>
++      <service_fmri value='svc:/system/filesystem/local'/>
++    </dependency>
++    <dependency name='net-loopback' grouping='require_all' restart_on='none' type='service'>
++      <service_fmri value='svc:/network/loopback'/>
++    </dependency>
++    <dependency name='net-physical' grouping='require_all' restart_on='none' type='service'>
++      <service_fmri value='svc:/network/physical'/>
++    </dependency>
++    <dependent name='p0f-daemon_multi-user' grouping='optional_all' restart_on='none'>
++      <service_fmri value='svc:/milestone/multi-user'/>
++    </dependent>
++<!--
++    ### The block below allows a client of p0f, such as milter-greylist, to
++    ### be started only after the p0f daemon successfully initializes, if the
++    ### p0f-daemon service is at all enabled (optional_all), otherwise this
++    ### particular client starts up unimpeded without p0f (and fails to query
++    ### the fingerprinter, but otherwise just works).
++-->
++<!--
++    <dependent name='milter-greylist' grouping='optional_all' restart_on='none'>
++      <service_fmri value='svc:/network/milter-greylist:default'/>
++    </dependent>
++-->
++    <exec_method name='start' type='method' exec='/usr/sbin/p0f.sh %m' timeout_seconds='120'>
++      <method_context>
++            <method_credential
++            user='root'
++            group='root'
++            limit_privileges=':default'
++	    privileges='basic,net_icmpaccess,net_rawaccess,file_chown,file_owner,file_dac_write,proc_setid,proc_chroot'/>
++<!--   ### The set of privileges above allows the SMF-ized daemon to access
++       ### the low-level networking. Normally SMF services are prohibited
++       ### from such intimacy with the OS in order to improve security.
++       ### Note we start as "root" above and later drop privileges in script. -->
++      </method_context>
++    </exec_method>
++    <exec_method name='stop' type='method' exec=':kill' timeout_seconds='60'>
++      <method_context/>
++    </exec_method>
++    <exec_method name='refresh' type='method' exec=':kill' timeout_seconds='180'>
++      <method_context/>
++    </exec_method>
++    <property_group name='general' type='framework'>
```

Comment:
Need what? The service? It is what we need this package for :)

p0f runs on the host (may be in local zone, including both exclusive and shared stacks) and sniffs the communications, such as incoming packets. Then it evaluates what OS fingerprint matches the remote IP address in the dialog, and caches the verdict. Then a p0f-client, such as milter-greylist, or its command-line tool, comes and asks "what OS is on that IP?" - and p0f already knows the answer. And then, in case of e.g. milter, we can be more discriminative against say random Windows desktops directly submitting e-mails to our relay (likely botnet drones).

So yes, by design and purpose it is a long-sunning sniffer service :)